internal/command/program_resolver_test.go

```diff
@@ -2,205 +2,264 @@ package command
+			program:         `export TEST_PARAM_EXPR=${TEST:7:0}`,
+			result:          &ProgramResolverResult{},
+			modifiedProgram: "export TEST_PARAM_EXPR=${TEST:7:0}\n",
 		},
 		{
-			name:   "arithmetic expression",
-			data:   `export TEST_ARITHM_EXPR=$(($z+3))`,
-			result: nil,
-			script: "export TEST_ARITHM_EXPR=$(($z + 3))\n",
+			name:            "arithmetic expression",
+			program:         `export TEST_ARITHM_EXPR=$(($z+3))`,
+			result:          &ProgramResolverResult{},
+			modifiedProgram: "export TEST_ARITHM_EXPR=$(($z + 3))\n",
 		},
 		{
-			name:   "value expression",
-			data:   `export TEST_VALUE_EXPR=$(echo -n "value")`,
-			result: nil,
-			script: "export TEST_VALUE_EXPR=$(echo -n \"value\")\n",
+			name:            "value expression",
+			program:         `export TEST_VALUE_EXPR=$(echo -n "value")`,
+			result:          &ProgramResolverResult{},
+			modifiedProgram: "export TEST_VALUE_EXPR=$(echo -n \"value\")\n",
 		},
 		{
-			name:   "double quoted value expression",
-			data:   `export TEST_DBL_QUOTE_VALUE_EXPR="$(echo -n 'value')"`,
-			result: nil,
-			script: "export TEST_DBL_QUOTE_VALUE_EXPR=\"$(echo -n 'value')\"\n",
+			name:            "double quoted value expression",
+			program:         `export TEST_DBL_QUOTE_VALUE_EXPR="$(echo -n 'value')"`,
+			result:          &ProgramResolverResult{},
+			modifiedProgram: "export TEST_DBL_QUOTE_VALUE_EXPR=\"$(echo -n 'value')\"\n",
 		},
 		{
-			name:   "default value",
-			data:   `export TEST_DEFAULT_VALUE=${TEST_DEFAULT_VALUE:-value}`,
-			result: nil,
-			script: "export TEST_DEFAULT_VALUE=${TEST_DEFAULT_VALUE:-value}\n",
+			name:            "default value",
+			program:         `export TEST_DEFAULT_VALUE=${TEST_DEFAULT_VALUE:-value}`,
+			result:          &ProgramResolverResult{},
+			modifiedProgram: "export TEST_DEFAULT_VALUE=${TEST_DEFAULT_VALUE:-value}\n",
 		},
 	}
 
-	for _, tc := range fixtures {
-		t.Run(
-			fmt.Sprintf("Auto_resolve %s", tc.name),
-			runProgramResolverFixtureTestCase(tc, ProgramResolverModeAuto),
-		)
-	}
+	for _, tc := range testCases {
+		t.Run("ProgramResolverModeAuto_"+tc.name, func(t *testing.T) {
+			r := NewProgramResolver(ProgramResolverModeAuto)
+			buf := bytes.NewBuffer(nil)
+			got, err := r.Resolve(strings.NewReader(tc.program), buf)
+			require.NoError(t, err)
+			assert.EqualValues(t, tc.modifiedProgram, buf.String())
+			assert.EqualValues(t, tc.result, got)
+		})
 
-	for _, tc := range fixtures {
-		t.Run(
-			fmt.Sprintf("Skip_resolve %s", tc.name),
-			runProgramResolverFixtureTestCase(tc, ProgramResolverModeSkip),
-		)
+		t.Run("ProgramResolverModeSkip_"+tc.name, func(t *testing.T) {
+			r := NewProgramResolver(ProgramResolverModeSkipAll)
+			buf := bytes.NewBuffer(nil)
+			got, err := r.Resolve(strings.NewReader(tc.program), buf)
+			require.NoError(t, err)
+			assert.EqualValues(t, tc.modifiedProgram, buf.String())
+			// In skip mode, all variables will be resolved.
+			if tc.result != nil {
```

Comment:
@sourishkrout I hope I figured this out correctly: when `ProgramResolverModeSkipAll`, then all variables are marked as resolved and their resolved value is equal to the original value.